### PR TITLE
Add an npm continuous deployment guide

### PR DESCRIPTION
**Pull request by dtinth**
At @taskworld, we deploy packages to npm from CircleCI. I wanted to write a guide about it, but I think it may be more beneficial to just put it in CircleCI docs 😄 

p.s. [`npm` should be spelled using lowercase](https://github.com/google/shipshape/blob/master/third_party/node/lib/node_modules/npm/doc/misc/npm-faq.md#is-it-npm-or-npm-or-npm)

p.s.2. or is it called continuous __delivery__?

### CODE REVIEW

File: jekyll/_docs/npm-continuous-deployment.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+layout: classic-docs
+title: "Continuous deployment with npm"
+short-title: "npm publish from CircleCI"
```
Comment (felicianotech) [inline jekyll/_docs/npm-continuous-deployment.md]: The "short title" here isn't really shorter. I think we can just leave it off completely.

File: jekyll/_docs/npm-continuous-deployment.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+description: "How to configure CircleCI to publish packages to npm automatically"
```
Comment (felicianotech) [inline jekyll/_docs/npm-continuous-deployment.md]: Wouldn't it be more accurate to say, publish to npmjs.org?

File: jekyll/_docs/npm-continuous-deployment.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    You can do that by logging in to npm (`npm login`). This will save the
+    authToken to the `~/.npmrc` file. Look for the following line:
+
+    ```
+    //registry.npmjs.org/:_authToken=00000000-0000-0000-0000-000000000000
+    ```
+
+    In this case, the authToken is `00000000-0000-0000-0000-000000000000`.
+
+2.  Go to your project settings, and set the `NPM_TOKEN` variable to the
```
Comment (felicianotech) [inline jekyll/_docs/npm-continuous-deployment.md]: Link "project settings" to https://circleci.com/docs/environment-variables/#setting-environment-variables-for-all-commands-without-adding-them-to-git. Proper Jekyll syntax would be `{{site.baseurl}}environment-variables/#setting-environment-variables-for-all-commands-without-adding-them-to-git`.

File: jekyll/_docs/npm-continuous-deployment.md (hunk 2 above)
Comment (dtinth) [inline jekyll/_docs/npm-continuous-deployment.md]: npm here refers to the npm registry, but I agree people may be confused. I’ll use “npmjs.org” to be more precise.

**Comment by dtinth** [comment]
Thanks for the review, updated as requested.

**Comment by dtinth** [comment]
I’ve [allowed edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/). If it’s more convenient for you, you can [edit the page content on my branch directly](https://github.com/taskworld/circleci-docs/edit/%EF%BD%8E%EF%BD%90%EF%BD%8D/jekyll/_docs/npm-continuous-deployment.md?pr=/circleci/circleci-docs/pull/578). :)

**Comment by felicianotech** [comment]
I had a few minor changes, but GitHub wasn't letting me save them because your branch name doesn't seem to be using ASCII characters and GitHub was 404ing. I'll do them in a new PR. Thank you for the contribution.

**Pull request merged**